ts/pages/account/activity.tsx

```diff
@@ -8,84 +8,124 @@ import { Breadcrumb } from 'ts/components/ui/breadcrumb';
 import { Table } from 'ts/components/ui/table';
 import { StakeStatus } from 'ts/components/ui/table_stake_status';
 
+import { useDispatch, useSelector } from 'react-redux';
+import { CallToAction } from 'ts/components/call_to_action';
+import { Dispatcher } from 'ts/redux/dispatcher';
+import { State } from 'ts/redux/reducer';
 import { colors } from 'ts/style/colors';
+import { AccountReady, EtherscanLinkSuffixes, StakingAPIDelegatorHistoryItem } from 'ts/types';
+import { formatEther, formatZrx } from 'ts/utils/format_number';
 import { utils } from 'ts/utils/utils';
 
-interface MockData {
-    timestamp: string | number;
-    date: string | number;
-    statusId: string | number;
-    amount: string | number;
-    status: string;
-    id: string;
+import { logUtils } from '@0x/utils';
+import { useAPIClient } from 'ts/hooks/use_api_client';
+import { errorReporter } from 'ts/utils/error_reporter';
+
+export interface ActivityProps {}
+
+const columns = ['Event timestamp', 'Event', 'Transaction'];
+
+function parseEvent(event: StakingAPIDelegatorHistoryItem): {title: string; subtitle: string} {
+    if (event.eventType === 'earned_rewards') {
+        const title = 'Earned Rewards';
+        const subtitle = `You earned ${formatEther(event.eventArgs.reward).formatted} ETH from pool ${event.eventArgs.poolId} for epoch ${event.eventArgs.epochId}.`;
+        return {title, subtitle};
+    } else if (event.eventType === 'deposited_zrx') {
+        const title = 'Deposited ZRX';
+        const subtitle = `You deposited ${formatZrx(event.eventArgs.zrxAmount).formatted} ZRX into the staking contract.`;
+        return {title, subtitle};
+    } else if (event.eventType === 'withdrew_zrx') {
+        const title = 'Withdrew ZRX';
+        const subtitle = `You withdrew ${formatZrx(event.eventArgs.zrxAmount).formatted} ZRX from the staking contract.`;
+        return {title, subtitle};
+    } else if (event.eventType === 'staked') {
+        const title = 'Staked';
+        const subtitle = `You staked ${formatZrx(event.eventArgs.zrxAmount).formatted} ZRX with pool ${event.eventArgs.poolId}.`;
+        return {title, subtitle};
+    } else if (event.eventType === 'removed_staked') {
+        const title = 'Removed Stake';
+        const subtitle = `You removed ${formatZrx(event.eventArgs.zrxAmount).formatted} ZRX from pool ${event.eventArgs.poolId}.`;
+        return {title, subtitle};
+    } else if (event.eventType === 'moved_stake') {
```

Comment:
This should be on the API level but I tested this out locally and it seems if I unstake by moveStake to `NILL_POOL_ID` it results in an `Unkown Event` whereas if I move from one pool to another it is correctly classified as `Moved Stake`.

![image](https://user-images.githubusercontent.com/5450545/79900876-37935600-840f-11ea-97db-168090717d5e.png)

Reply:
Sounds like an API fix. Can you send me the transaction hashes that generated unknown events?

Reply:
Ah it's bc I specified in the event parser `removed_staked` instead of `removed_stake` 🤦 . I'll fix it.

Reply:
Fixed!

Reply:
![image](https://user-images.githubusercontent.com/13876978/79904033-ec565480-83c8-11ea-80bf-94d7842e78ac.png)